src/viking.js

```diff
@@ -1,11 +1,94 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`;
+    } else return `${this.name} has died in act of combat`;
+  }
+
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`;
+    } else return `A Saxon has died in combat`;
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+
+  addViking(aVikingObject) {
+    this.vikingArmy.push(aVikingObject);
+  }
+
+  addSaxon(aSaxonObject) {
+    this.saxonArmy.push(aSaxonObject);
+  }
+
+  vikingAttack() {
+    return this.attack(this.vikingArmy, this.saxonArmy);
+  }
+
+  saxonAttack() {
+    return this.attack(this.saxonArmy, this.vikingArmy);
+  }
+
+  removeDeadSoldier(whatArmy) {
```

Comment:
this is a neat little function. :) 
you will love filter, because it does that but much easier. also this might not behave as you think it would in some edge cases, because you change the length of the array and shift around indeces as you do so, so generally when you mutate an array, while you are still iterating over it, that can lead to subtle bugs that are hard to catch.